AK/StringView.h

```diff
@@ -171,12 +172,12 @@ class StringView {
         while (maybe_separator_index.has_value()) {
             auto separator_index = maybe_separator_index.value();
             auto part_with_separator = view.substring_view(0, separator_index + separator.length());
-            if (keep_empty || separator_index > 0)
+            if (has_flag(action_on_split, SplitBehavior::KeepEmpty) || separator_index > 0)
                 callback(part_with_separator.substring_view(0, separator_index));
             view = view.substring_view_starting_after_substring(part_with_separator);
             maybe_separator_index = view.find(separator);
         }
-        if (keep_empty || !view.is_empty())
+        if (has_flag(action_on_split, SplitBehavior::KeepEmpty) || !view.is_empty())
```

Comment:
I wrote the notes about the KeepSeparator flag before I wrote, this, but the same applies: Calculate `has_flag()` once before the loop, and use the result in these places, instead of calculating it repeatedly.